### DISCUSSION TITLE
Get a Portable Version of Oh My Posh

### DISCUSSION THREAD

**Discussion by RoyiAvital** [JanDeDobbeleer/oh-my-posh#4749] [General]
Hi,

Is there a way to get a portable version of Oh My Posh?

If I get this correctly, basically all needed is the `exe` file on path and use the themes.
Yet I couldn't extract the `exe` from the Windows binary using `7Zip`.

Any other way to get it?

**Comment by JanDeDobbeleer** [comment]
You only need the exe, yes. Not even the themes, completely optional.

**Reply by RoyiAvital** [reply]
But how can one get it? Or is it just the `exe` in releases?
I assume on Releases it is the installer.

**Reply by JanDeDobbeleer** [reply]
@RoyiAvital it's both the installer and the `posh-windows-arch.exe` that's available. You need the latter.

**Reply by RoyiAvital** [reply]
So I need to have `posh-windows-amd64.exe` on my `$env:PATH`, this is basically what's done by the installer?

**Reply by JanDeDobbeleer** [reply]
Correct. But then rename that to oh-my-posh, or use this executable's name.

**Reply by RoyiAvital** [reply]
OK, I see that the way to actiavet it is by:

```powershell
oh-my-posh init pwsh --config ~/jandedobbeleer.omp.json | Invoke-Expression
```

This one changes the profile script.
Is there a way to get the code to run for this session only without changing the profile file?

The motivation is creating a portable environment with Oh My Posh.

**Reply by ATOMPEAR** [reply]
`oh-my-posh init pwsh --config ~/jandedobbeleer.omp.json | Invoke-Expression`

Is this powershell? Will that add it to path? This is so confusing. I want it to be portable. Because I have windows terminal portable distributed with my app. Which I want to be styles by either oh-my-posh or startship. IDK which one would be better?

**Reply by JanDeDobbeleer** [reply]
@ATOMPEAR no this will not add it to the `PATH`. An executable needs to be added to the PATH prior to being invoked, otherwise the shell will say it can't find it. `PATH` management is device management, something the installer does for you, but when you use the standalone executable, that's work you need to do.

**Reply by RoyiAvital** [reply]
@JanDeDobbeleer , Is there a way to have OhMyPosh without touching the user PS profile file?

**Reply by ATOMPEAR** [reply]
@JanDeDobbeleer I really over complicated this apparently aha I got it I think. Im still confused about the $PROFILE nothing is clicking as to what that is a reference to.

**Reply by JanDeDobbeleer** [reply]
@RoyiAvital pwsh.exe doesn't have a way to add code to be executed on startup, that's why you have the `$PROFILE`, just like you have a `bashrc`, `zshrc` etc...

**Reply by JanDeDobbeleer** [reply]
@ATOMPEAR https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_profiles?view=powershell-7.4

**Reply by RoyiAvital** [reply]
@JanDeDobbeleer , I am not sure I understand. I want to enable it for a single session. Hence I don't want it to change my Profile. I have it on my `PATH`. Is there a way to do so?

**Reply by JanDeDobbeleer** [reply]
@RoyiAvital not that I'm aware off. Unless your terminal has a way to execute the init line when starting the terminal, the profile is the only way.

**Reply by RoyiAvital** [reply]
@JanDeDobbeleer , I can run the init command.  btu doesn't the init command change the ps profile file?

**Reply by JanDeDobbeleer** [reply]
@RoyiAvital the init command outputs a PowerShell script (or another depending on the shell you initialize). It needs to be run from the shell you're trying to set up with oh-my-posh.

**Reply by lewis-yeung** [reply]
> doesn't the init command change the ps profile file?

@RoyiAvital No, it doesn't. The command just does initialization each time a session starts.
